src/engine/source/indexerconnector/include/indexerConnector/indexerConnector.hpp

```diff
@@ -71,6 +74,7 @@ class EXPORTED IndexerConnector final : public IIndexerConnector
     std::atomic<bool> m_stopping {false};
     std::string m_indexName;
     std::mutex m_syncMutex;
+    std::vector<nlohmann::json> m_processedEvents;
```

Comment:
Adding this as a member of the class is very insecure (remember that the MULTI_THREADED_UNORDERED mode means many threads using the lambda, therefore you have to be thread safe), this could be as a local variable inside the while.

Reply:
Now the array is declared in the lambda.
Also, some copy/dump extra operations were removed to improve the performance.